File: src/main/java/net/punchtree/loquainteractable/displayutil/ArmorStandChunkLoadingReglow.java
package net.punchtree.loquainteractable.displayutil;

import org.bukkit.entity.ArmorStand;
import org.bukkit.entity.Entity;
import org.bukkit.entity.EntityType;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.world.ChunkLoadEvent;
import org.bukkit.scheduler.BukkitRunnable;

import net.punchtree.loquainteractable.LoquaInteractablePlugin;

public class ArmorStandChunkLoadingReglow implements Listener {

	@EventHandler
	public void onChunkLoad(ChunkLoadEvent event) {
		// TODO move this into latching functions onto one loop off the main plugin instance?
		for (Entity entity : event.getChunk().getEntities()) {
			if (entity.getType() == EntityType.ARMOR_STAND) {
				ArmorStand stand = (ArmorStand) entity;
				if (stand.getScoreboardTags().contains("loqinttempglowing")) {
					stand.setCanTick(true);
					stand.setFireTicks(1000);
					new BukkitRunnable() {
						public void run() {							
							stand.setCanTick(false);
						}
					}.runTaskLater(LoquaInteractablePlugin.getInstance(), 20);
				}
			}
		}
	}
	
}

File: src/main/java/net/punchtree/loquainteractable/displayutil/ArmorStandChunkLoadingReglow.kt
package net.punchtree.loquainteractable.displayutil

import net.punchtree.loquainteractable.LoquaInteractablePlugin
import org.bukkit.entity.ArmorStand
import org.bukkit.entity.EntityType
import org.bukkit.event.Listener
import org.bukkit.event.world.ChunkLoadEvent
import org.bukkit.scheduler.BukkitRunnable

object ArmorStandChunkLoadingReglow : Listener {

    internal fun doArmorStandReglow(event: ChunkLoadEvent) {
        // TODO move this into latching functions onto one loop off the main plugin instance?
        for (entity in event.getChunk().entities.filter {
            it.type == EntityType.ARMOR_STAND
                    && it.scoreboardTags.contains("loqinttempglowing")
        }) {
            val stand = entity as ArmorStand
            stand.setCanTick(true)
            stand.setFireTicks(1000)
            object : BukkitRunnable() {
                override fun run() {
                    stand.setCanTick(false)
                }
            }.runTaskLater(LoquaInteractablePlugin.instance, 20)
        }
    }

}
